sentences/DPT-surface.js

```diff
@@ -0,0 +1,26 @@
+/**
+Depth:
+$IIDPT,x.x,x.x,,*hh
+ I I_Sensor offset, >0 = surface transducer distance, >0 = keel transducer distance.
+ I_From Surface To Transduder
+
+ */
+// NMEA0183 Encoder DPT   $IIDPT,69.21,-0.001*60
+const nmea = require('../nmea.js')
+module.exports = function (app) {
+  return {
+    sentence: 'DPT',
+    title: 'DPT - Depth at Surface (using surfaceToTransducer)',
+    keys: [
+      'environment.depth.belowTransducer',
+      'environment.depth.surfaceToTransducer'
+    ],
+    f: function dpt (belowTransducer, surfaceToTransducer) {
+      return nmea.toSentence([
+        '$IIDPT',
+        belowTransducer.toFixed(2),
+        surfaceToTransducer.toFixed(3)
```

Comment:
Since both surfaceToTransducer and are positive/absolute values in the SK spec either this or the pre-existing DPT is incorrect https://github.com/SignalK/signalk-to-nmea0183/blob/master/sentences/DPT.js#L22

Which is correct?

This relates also to the comment in code, where it says "greater than" for both values.

Reply:
dervived-data plugins return a negative value for transducerToKeel calculation so the DPT.py works with it's neg value.  If that plugin gets update then we'll want to update DPT.js code as well.

DPT 2nd field:  Offset from transducer, meters -->  positive means distance from transducer to water line, negative means distance from transducer to keel

Reply:
In signalk-derived-data plugin calcs/transducerToKeel.js calculates this:
const transducerToKeel = surfaceToTransducer - draft

Reply:
Which will return a negative number not a positive/absolute number as you mention the SK spec.  This means the current DPT always returns, correctly, a negative value - the offset value from transducer to keel.  So we can not touch either and still have the correct negative offset or make sure derived-data plugin returns the absolute value and then update DPT.js to flip the sign on this value.  I'd be happy to do a PR for both if you like.   DPT-surface.js generate the other form of this sentence w/ a positive offset, ie. distance from surface to transducer.

Reply:
Yeah, I saw. I think we got that wrong (with either too ambiguous specification or incorrect formula in derived-data). 

I am not eager to change derived data, but I think dpt.js could use a fix here so that it forces the value to be negative.

Reply:
Maybe you already saw https://github.com/SignalK/signalk-to-nmea0183/pull/103